### ISSUE TITLE
wrong ordering GPU and enumeration GPU ID

### ISSUE THREAD

**Issue by khunpoum** [RadeonOpenCompute/ROC-smi#45]
in list have  GPU1 - its internal intel graphics.
how i can exclude non AMD cards from output? 
because i confused - other soft make numeration of GPU not the same.

Ubuntu 16.04, AMDGPU-pro 17.40

here in list must be 10 GPU with GPU ID from 0 to 9 such as in all other soft like atiflash and miner softs

````
ROC-smi-roc-1.9.1# ./rocm-smi 

====================    ROCm System Management Interface    ====================
================================================================================
 GPU  Temp    AvgPwr   SCLK     MCLK     Fan      Perf    SCLK OD    MCLK OD
  0   55c     117.161W 1215Mhz  2080Mhz  40.0%    manual    0%         0%       
  1   N/A     N/A      N/A      N/A      0%       N/A       N/A        N/A      
  10  56c     115.118W 1215Mhz  1980Mhz  40.0%    manual    0%         0%       
  2   57c     118.4W   1215Mhz  2020Mhz  40.0%    manual    0%         0%       
  3   58c     115.139W 1200Mhz  2000Mhz  23.92%   manual    0%         0%       
  4   41c     72.120W  1200Mhz  300Mhz   17.65%   manual    0%         0%       
  5   53c     113.128W 1215Mhz  2060Mhz  40.0%    manual    0%         0%       
  6   57c     119.55W  1215Mhz  2020Mhz  40.0%    manual    0%         0%       
  7   53c     111.198W 1215Mhz  2080Mhz  40.0%    manual    0%         0%       
  8   58c     115.231W 1200Mhz  2000Mhz  21.96%   manual    0%         0%       
  9   55c     113.60W  1200Mhz  2000Mhz  40.0%    manual    0%         0%       
================================================================================
```

**Comment by kentrussell** [comment]
The SMI just polls the GPUs from the /sys/class/drm/ folder. This includes non-AMD GPUs, which is why it comes up like that. The SMI would need a rework to exclude those other GPUs. Plus, there are some GPUs that support some functionality (like only fan control), but not all functionality, which makes it a little tougher. It's something that can be done later on, but isn't a critical priority at this time. I'll add it to the list of QOL changes though.

**Comment by kentrussell** [comment]
I have a patch in 2.2 to address this issue. I'll close this once that is released

Edit: I hoped to get it into 2.1 but missed the cutoff. I can upload it to krussell/fixes if you want to take a look

**Comment by kentrussell** [comment]
Issue resolved in 2.2 release

**Comment by jlgreathouse** [comment]
Hi @kentrussell -- does this mean we can also close out RadeonOpenCompute/ROCM#561 ?
